src/dns.ts

```diff
@@ -72,6 +78,14 @@ export class DNS implements DNSInterface {
           types
         })
 
+        result.Answer = result.Answer
+          .map((answer) => {
+            // convert type to either RecordType or RecordTypeLabel
+            answer.type = convertType(answer.type, options.useRecordTypeValue)
+
+            return answer
+          })
+
         for (const answer of result.Answer) {
           this.cache.add(domain, answer)
         }
```

Comment:
If we implement https://github.com/multiformats/js-dns/pull/5#discussion_r1532917413, we should change this to 

```suggestion
        for (const answer of result.Answer) {
          this.cache.add(domain, answer)
        }
        
        result.Answer = result.Answer
          .map((answer) => {
            // convert type to either RecordType or RecordTypeLabel
            answer.type = convertType(answer.type, options.useRecordTypeValue)

            return answer
          })
```